<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/erle/ros2_mara_ws/src/mara/mara_description/urdf/mara_robot_camera_top.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="mara" xmlns:xacro="http://ros.org/wiki/xacro">
  <gazebo>
    <plugin name="mara_joint_state" filename="libgazebo_ros_joint_state_publisher.so">
		<ros>
			<!-- <namespace>/mara</namespace> -->
			<argument>~/out:=joint_states</argument>
		</ros>
		<update_rate>250</update_rate>
		<joint_name>motor1</joint_name>
		<joint_name>motor2</joint_name>
		<joint_name>motor3</joint_name>
		<joint_name>motor4</joint_name>
		<joint_name>motor5</joint_name>
    <joint_name>motor6</joint_name>
	</plugin>

  <plugin name="hros_actuation_servomotor_000000000001" filename="libmara_gazebo_joint_plugin.so">
    <ros>
      <!-- <namespace>/tb3</namespace> -->
    </ros>

    <axis1>motor2</axis1>
    <axis2>motor1</axis2>
  </plugin>

  <plugin name="hros_actuation_servomotor_000000000002" filename="libmara_gazebo_joint_plugin.so">
    <ros>
      <!-- <namespace>/tb3</namespace> -->
    </ros>

    <axis1>motor3</axis1>
    <axis2>motor4</axis2>
  </plugin>

  <plugin name="hros_actuation_servomotor_000000000003" filename="libmara_gazebo_joint_plugin.so">
    <ros>
      <!-- <namespace>/tb3</namespace> -->
    </ros>

    <axis1>motor5</axis1>
    <axis2>motor6</axis2>
  </plugin>
  </gazebo>
  <!--
  Author: Alejandro  Hernandez
-->
  <!-- <xacro:include filename="$(find mara_description)/urdf/camera_top.urdf.xacro" /> -->
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="8.625E-08 -4.6583E-06 0.03145"/>
      <mass value="7.033"/>
      <inertia ixx="0.1478344" ixy="-0.00001" ixz="0.0" iyy="0.14782335" iyz="0.0" izz="0.29471799"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_BASE.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Gazebo/Grey">
        <color rgba="0. 0. 0. 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_BASE.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="base_link">
    <material>Gazebo/Grey</material>
    <selfCollide>True</selfCollide>
  </gazebo>
  <link name="motor1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="8.625E-08 -4.6583E-06 0.03145"/>
      <mass value="5.8"/>
      <inertia ixx="0.10105893" ixy="0.000211" ixz="-0.004" iyy="0.09386624" iyz="-0.01698537" izz="0.01449937"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_mara1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Gazebo/White">
        <color rgba="0. 0. 0. 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_mara1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="motor1_link">
    <material>Gazebo/White</material>
    <selfCollide>True</selfCollide>
  </gazebo>
  <link name="motor2_link">
    <inertial>
      <origin rpy="0 0 0" xyz="8.625E-08 -4.6583E-06 0.03145"/>
      <mass value="1.03237"/>
      <inertia ixx="0.014865226" ixy="0.0" ixz="0.0" iyy="0.01557" iyz="0.00071" izz="0.00092"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_LINK1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Gazebo/Black">
        <color rgba="0. 0. 0. 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_LINK1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="motor2_link">
    <material>Gazebo/Black</material>
    <selfCollide>True</selfCollide>
  </gazebo>
  <link name="motor3_link">
    <inertial>
      <origin rpy="0 0 0" xyz="8.625E-08 -4.6583E-06 0.03145"/>
      <mass value="3.8"/>
      <inertia ixx="0.04517" ixy="0.00002" ixz="0.0" iyy="0.01587" iyz="0.01444" izz="0.03303"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_mara2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Gazebo/White">
        <color rgba="0. 0. 0. 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_mara2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="motor3_link">
    <material>Gazebo/White</material>
    <selfCollide>True</selfCollide>
  </gazebo>
  <link name="motor4_link">
    <inertial>
      <origin rpy="0 0 0" xyz="8.625E-08 -4.6583E-06 0.03145"/>
      <mass value="1.1"/>
      <inertia ixx="0.01464" ixy="0.0" ixz="0.0" iyy="0.00895" iyz="-0.00631" izz="0.00725"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_LINK2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Gazebo/Black">
        <color rgba="0. 0. 0. 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_LINK2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="motor4_link">
    <material>Gazebo/Black</material>
    <selfCollide>True</selfCollide>
  </gazebo>
  <link name="motor5_link">
    <inertial>
      <origin rpy="0 0 0" xyz="8.625E-08 -4.6583E-06 0.03145"/>
      <mass value="2.9"/>
      <inertia ixx="0.03406" ixy="0.0" ixz="0.0" iyy="0.01066" iyz="0.00989" izz="0.02563"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_mara3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Gazebo/White">
        <color rgba="0. 0. 0. 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/H-ROS_Robot_mara3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="motor5_link">
    <material>Gazebo/White</material>
    <selfCollide>True</selfCollide>
  </gazebo>
  <link name="motor6_link">
    <inertial>
      <origin rpy="0 0 0" xyz="8.625E-08 -4.6583E-06 0.03145"/>
      <mass value="2.9"/>
      <inertia ixx="0.0005" ixy="0.0" ixz="0.0" iyy="0.0005" iyz="0.0000" izz="0.0003"/>
    </inertial>
    <visual>
      <origin rpy="0 0 1.04666666667" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/gripper_support.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Gazebo/Black">
        <color rgba="0. 0. 0. 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 1.04666666667" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mara_description/meshes/gripper_support.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="motor6_link">
    <material>Gazebo/Black</material>
    <selfCollide>True</selfCollide>
  </gazebo>
  <link name="ee_link">
    </link>
  <gazebo reference="ee_link">
    <material>Gazebo/Blue</material>
    <selfCollide>False</selfCollide>
  </gazebo>
  <joint name="motor1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.00625"/>
    <parent link="base_link"/>
    <child link="motor1_link"/>
    <axis xyz="0 0 -1"/>
    <limit effort="330.0" lower="-2.3562" upper="2.3562" velocity="3.1416"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="motor2" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.100499999 0.1545"/>
    <parent link="motor1_link"/>
    <child link="motor2_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="330.0" lower="-3.13" upper="3.13" velocity="3.1416"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="motor3" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.000499999 0.2"/>
    <parent link="motor2_link"/>
    <child link="motor3_link"/>
    <axis xyz="0 -1 0"/>
    <limit effort="330" lower="-3.13" upper="3.13" velocity="3.1416"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="motor4" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.09999 0.12999997"/>
    <parent link="motor3_link"/>
    <child link="motor4_link"/>
    <axis xyz="0 0 -1"/>
    <limit effort="330" lower="-3.13" upper="3.13" velocity="3.1416"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="motor5" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.096999 0.15906"/>
    <parent link="motor4_link"/>
    <child link="motor5_link"/>
    <axis xyz="0 -1 0"/>
    <limit effort="330" lower="-3.13" upper="3.13" velocity="3.1416"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="motor6" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.0969999 0.1265"/>
    <parent link="motor5_link"/>
    <child link="motor6_link"/>
    <axis xyz="0 0 -1"/>
    <limit effort="330" lower="-3.13" upper="3.13" velocity="3.1416"/>
    <dynamics damping="0" friction="0"/>
  </joint>
  <joint name="ee_link" type="fixed">
    <origin rpy="0 -1.57 0" xyz="0 0.0 0.24"/>
    <parent link="motor6_link"/>
    <child link="ee_link"/>
  </joint>
  <transmission name="shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- nothing to do here at the moment -->
  <!-- ROS ${prefix}base_link to UR 'Base' Coordinates transform -->
  <link name="base"/>
  <joint name="base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as ${prefix}base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="tool0"/>
  <joint name="motor6_link-tool0_fixed_joint" type="fixed">
    <origin rpy="0 -1.57 0" xyz="0 0.0 0.24"/>
    <parent link="motor6_link"/>
    <child link="tool0"/>
  </joint>
  <link name="table">
    <contact>
      <lateral_friction value="1.0"/>
    </contact>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".0"/>
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.6"/>
      <geometry>
        <mesh filename="package://mara_description/urdf/table.obj" scale="1.5 1 0.05"/>
      </geometry>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.6"/>
      <geometry>
        <box size="1.5 1 0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="-0.65 -0.4 0.29"/>
      <geometry>
        <mesh filename="package://mara_description/urdf/table.obj" scale="0.1 0.1 0.58"/>
      </geometry>
      <material name="framemat0"/>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="-0.65 0.4 0.29"/>
      <geometry>
        <mesh filename="package://mara_description/urdf/table.obj" scale="0.1 0.1 0.58"/>
      </geometry>
      <material name="framemat0"/>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0.65 -0.4 0.29"/>
      <geometry>
        <mesh filename="package://mara_description/urdf/table.obj" scale="0.1 0.1 0.58"/>
      </geometry>
      <material name="framemat0"/>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0.65 0.4 0.29"/>
      <geometry>
        <mesh filename="package://mara_description/urdf/table.obj" scale="0.1 0.1 0.58"/>
      </geometry>
    </visual>
  </link>
  <link name="world"/>
  <!-- <joint name="end_effector_adapter_to_robotiq" type="fixed">
    <origin rpy="0 0 0" xyz=" 0 0 0.05375"/>
    <parent link="motor6_link"/>
    <child link="robotiq_arg2f_base_link"/>
  </joint> -->
  <joint name="world_to_table_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.02"/>
    <parent link="world"/>
    <child link="table"/>
  </joint>
  <joint name="table_to_base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.65"/>
    <parent link="table"/>
    <child link="base_link"/>
  </joint>
</robot>
